### ISSUE TITLE
Make server URL configurable

### ISSUE THREAD

**Issue by schiessle** [cleverdevil/news_indieweb#1]
If I see it correctly, the server URL is hard coded in the JS file. It would be nice to have a personal setting for it. This way also different users can publish to different servers.

I can have a look at it, if you don't already work in it.

**Comment by schiessle** [comment]
Actually it would be nice to have a option to enter the URL in the News app settings but I assume there is no way for plugins to add something to the settings, right @BernhardPosselt

**Comment by BernhardPosselt** [comment]
I dont really get the question :)

**Comment by martymcguire** [comment]
Would the app settings API do what you need?

https://docs.nextcloud.com/server/10/developer_manual/app/settings.html

**Comment by BernhardPosselt** [comment]
What is the goal?

**Comment by schiessle** [comment]
@BernhardPosselt at the moment the link to the Webpage which runs Known (https://withknown.com/) is hard coded in the JavaScript file. This means a admin needs to edit the source code to point to his instance of Known to use the Buttons created by @cleverdevil. It also has the draw back that each user can only interact with this one Known instance, defined in the source code. Of course that's not ideal, both that you can't set it per user and that you have to edit the source code.

I would expect that each user can configure his own Known instance. The app could now register a personal setting so that the user can configure his Known instance in the personal settings but it would be more natural if it could be configured in the News app. That's where my question comes from, is there a way to hook into the News app settings like you can hook into the personal settings? I fear I know the answer 😉

**Comment by BernhardPosselt** [comment]
If things like this are shipped in socialsharing it would probably make sense to configure it there :D

**Comment by BernhardPosselt** [comment]
You probably want to configure this for all instances

**Comment by schiessle** [comment]
@BernhardPosselt no, it doesn't has anything to do with the social sharing buttons I mentioned here https://github.com/nextcloud/news/issues/128

This one we are discussing here is a complete independent plugin. The question boils down to: If a plugin needs some configuration, where should it happen? Does the News app API provides a way to hook into the News settings or do each plugin needs to register a personal/admin settings page?

**Comment by BernhardPosselt** [comment]
I see, ideally there is no need for additional settings or settings are somehow hooked up in your profile settings. 

I mean you need the following things: PHP controller for validating & saving and setting values, js functionality to load defaults and change settings possibly validating input and changing state, html templates and a way to hook that up into the settings area. The more external APIs are present, the harder it is to migrate a new frontend architecture.

I mean ofc you could define a plugin API for that but if you want to have that implemented in a timely manner I'd recommend to look for a different solution, e.g. a configure button which creates a text input popup.

**Comment by cleverdevil** [comment]
So, a few notes:

1. The plugin originally (before I tossed it up on GitHub) talked directly to my [personal website](https://cleverdevil.io) running the [Known CMS](https://withknown.com). I then discovered that there was no (easy) way for a Nextcloud News plugin to expose additional configurable settings, so that other people could put in their own website.
2. As a result of #1, I changed things up to instead hardcode the sharing link to [Quill](https://quill.p3k.io). [Quill is open source](https://github.com/aaronpk/Quill), but the version that I am linking to is a free, hosted version. Quill implements [Micropub](https://micropub.net) and enables all of the interactions of this plugin to my personal website, which is Micropub compatible. This means that anyone can take the plugin as is and use it to get these interactions without any configuration, assuming that their website is Micropub compatible.
3. In the ideal world, Nextcloud would detect if my personal website as configured in my Nextcloud profile supports Micropub, [Webmention](https://www.w3.org/TR/webmention/), and other [IndieWeb](https://indieweb.org) protocols, and enable pervasive support across the board for any app or feature of Nextcloud. That said, my plugin is a good start for just my use case! As I mentioned, Quill is open source, so it could be a good starting point, and @aaronpk is generally happy to help spread the IndieWeb love.

Hope this helps clarify!
